### PR TITLE
Send mem and pid fds as well as notify fds

### PR DESCRIPTION
**Pull request by solarkennedy**
Although we could look into the tini process to find its fd from the socket, this won't work with the oci hook / kubelet method.

So instead I've adjusted this to do the same thing that lxc and what the oci hook would do.

I don't know how to "really" test this yet. Strace does say this was sent (nothing shows up in socat?):

    sendmsg(3, {msg_name=NULL, msg_namelen=0, msg_iov=[{iov_base="\0", iov_len=1}], msg_iovlen=1, msg_control=[{cmsg_len=28, cmsg_level=SOL_SOCKET, cmsg_type=SCM_RIGHTS, cmsg_data=[5, 6, 4]}], msg_controllen=28, msg_flags=0}, 0) = 1

cc @keerti-n

### CODE REVIEW

File: src/seccomp_fd_notify.c [hunk 1]
```diff
@@ -103,22 +115,41 @@ static int install_notify_filter(void) {
 	return notify_fd;
 }
 
+int get_mem_fd(int pid) {
+	char mem_path[6 /* /proc/ */
+		      + 22 /* INTTYPE_TO_STRLEN(int64_t) */
+		      + 3 /* mem */
+		      + 1 /* \0 */];
+	snprintf(mem_path, sizeof(mem_path), "/proc/%d/mem", pid);
```
Comment (sargun) [inline src/seccomp_fd_notify.c:120]: are you sure /proc is mounted (and valid) at this point?

File: src/seccomp_fd_notify.c [hunk 2]
```diff
@@ -103,22 +115,41 @@ static int install_notify_filter(void) {
...
+	int mem_fd = open(mem_path, O_RDWR | O_CLOEXEC);
+	return mem_fd;
+}
+
+int get_pid_fd(int pid) {
+	char pid_path[6 /* /proc/ */
+		      + 22 /* INTTYPE_TO_STRLEN(int64_t) */
```
Comment (sargun) [inline src/seccomp_fd_notify.c]: just use PATH_MAX.

**Review by sargun — APPROVED** [review]
I still think that we should do some validation on who's on the other end of the socket (ideally the same way titus executor does it).

File: src/seccomp_fd_notify.c (hunk 1 above)
Comment (solarkennedy) [reply src/seccomp_fd_notify.c:120]: I would be surprised if it wasn't? I would be surprised if containerd is spawning processes before the container is fully setup, including its mounts/filesystem.

**Comment by solarkennedy** [comment]
> I still think that we should do some validation on who's on the other end of the socket (ideally the same way titus executor does it).

Agreed. Otherwise this is kinda an RCE-as-a-service sorta.

**Pull request merged**